jackal_base/src/jackal_hardware.cpp

```diff
@@ -31,78 +32,235 @@
  *
  */
 
-#include <boost/assign.hpp>
-#include "jackal_base/jackal_hardware.h"
+#include "jackal_base/jackal_hardware.hpp"
+#include "jackal_msgs/msg/feedback.hpp"
+#include "hardware_interface/types/hardware_interface_type_values.hpp"
 
 namespace jackal_base
 {
 
-JackalHardware::JackalHardware()
+static const std::string HW_NAME = "JackalHardware";
+static const std::string LEFT_CMD_JOINT_NAME = "front_left_wheel_joint";
+static const std::string RIGHT_CMD_JOINT_NAME = "front_right_wheel_joint";
+
+/**
+ * @brief Write commanded velocities to the MCU
+ * 
+ */
+void JackalHardware::writeCommandsToHardware()
 {
-  ros::V_string joint_names = boost::assign::list_of("front_left_wheel")
-      ("front_right_wheel")("rear_left_wheel")("rear_right_wheel");
-
-  for (unsigned int i = 0; i < joint_names.size(); i++)
-  {
-    hardware_interface::JointStateHandle joint_state_handle(joint_names[i],
-        &joints_[i].position, &joints_[i].velocity, &joints_[i].effort);
-    joint_state_interface_.registerHandle(joint_state_handle);
-
-    hardware_interface::JointHandle joint_handle(
-        joint_state_handle, &joints_[i].velocity_command);
-    velocity_joint_interface_.registerHandle(joint_handle);
-  }
-  registerInterface(&joint_state_interface_);
-  registerInterface(&velocity_joint_interface_);
+  double diff_speed_left = hw_commands_[left_cmd_joint_index_];
+  double diff_speed_right = hw_commands_[right_cmd_joint_index_];
 
-  feedback_sub_ = nh_.subscribe("feedback", 1, &JackalHardware::feedbackCallback, this);
+  int8_t mode = jackal_msgs::msg::Drive::MODE_VELOCITY;
 
-  // Realtime publisher, initializes differently from regular ros::Publisher
-  cmd_drive_pub_.init(nh_, "cmd_drive", 1);
+  if (std::abs(diff_speed_left) < 0.01 && std::abs(diff_speed_right) < 0.01) {
```

Comment:
Will this stop the Jackal from holding zero velocity since the velocity controller will be turned off?

Reply:
Drive mode NONE automatically sets the wheel velocities to zero.